plugins/chromatic_correction.py

```diff
@@ -0,0 +1,346 @@
+from typing import Dict
+
+import numpy
+import wx
+import yaml
+from scipy.spatial.distance import cdist
+
+from odemis import model, dataio
+from odemis.acq.acqmng import acquireZStack
+from odemis.acq.stream import FluoStream
+from odemis.gui.plugin import Plugin
+from odemis.gui.util import get_home_folder
+from odemis.model import DataArray
+from odemis.util.comp import generate_zlevels
+from odemis.util.spot import MaximaFind
+from odemis.util.transform import AffineTransform
+
+# Below values are based on one experiment dataset, needs more testing
+# Diameter might need to be adapted if the sample or the objective is changed
+# Objective of 50x|0.5 na was used
+# Actual bead size used was 200nm
+ACTUAL_BEAD_DIAMETER = 200 * 1e-9  # m
+# Based on diffraction, bead size in the image appears bigger which is used in calculation for finding and matching
+# bead centers. The actual bead diameter is multiplied by a scaling factor
+# The scaling is based on the provided dataset
+SCALE = 5.5
+# Maximum number of beads to find in the given image
+BEAD_QUANTITY = 15
+
+
+def find_corresponding_points(centers_1: numpy.ndarray, centers_2: numpy.ndarray, max_distance: float):
+    """
+    Find corresponding points in two sets of points that are within a specified maximum distance.
+    :param centers_1: Nx2 array of points (x, y) in the first image.
+    :param centers_2: Mx2 array of points (x, y) in the second image.
+    :param max_distance: (float in pixels) Maximum allowable distance for point pairs.
+    :returns:
+        pairs: (numpy.ndarray) Corresponding point pairs (index1, index2).
+    """
+    # Calculate pairwise distances
+    distances = cdist(centers_1, centers_2)
+
+    # Find pairs within the max_distance
+    all_pairs = numpy.argwhere(distances <= max_distance)
+    sorted_pairs = sorted(all_pairs, key=lambda x: distances[x[0], x[1]])
+
+    matched_1 = set()
+    matched_2 = set()
+    pairs = []
+
+    for i, j in sorted_pairs:
+        if i not in matched_1 and j not in matched_2:
+            pairs.append([i, j])
+            matched_1.add(i)
+            matched_2.add(j)
+
+    return numpy.array(pairs)
+
+
+def estimate_transformation_from_pointset(points1: numpy.ndarray, points2: numpy.ndarray):
+    """
+    Estimate the affine transformation values from a set of corresponding points that maps `points2` to `points1`.
+    :param points1: Nx2 (x, y) location of features
+    :param points2: Nx2 (x, y) location of the corresponding features
+    :returns:
+        tuple: A tuple containing correction parameters:
+            - scale_cor ((float, float)): Scaling factors along the x and y axes
+            - translation_cor ((float, float)): Translation components along the x and y axes
+            - rotation_cor (float): Rotation angle (in radians)
+            - shear_cor (float): Shear angle (in radians)
+    """
+    affine = AffineTransform.from_pointset(points1, points2)
+    mat = numpy.eye(3)
```

Comment:
Similar to other comment, will close this one

Reply:
I think this is a serious problem and an indication that one of the two is incorrect. So far I've never had a problem with the AffineTransform module, so it might just be a different interpretation of shear/scale/squeeze. I am a bit hesitant to approve this PR until it's been figured out what the origin of the difference is.

Reply:
I fully agree that this function should be moved to `odemis.util`. Either in util.conversion or util.transform.